WebXRManager: Correctly update the user camera when it has a parent

rhuitl opened the pull request:
Related PRs: #21964 (initial attempt to address this issue, I suppose) and #22362 (revert of the former)

This is my attempt on fixing the issue. It works for my application, where I previously had to update the user camera manually after rendering, and it also seems to work for the roller coaster example (which was working before, but now is also working after this PR).

**Description**

This PR addresses the issue that the user camera pose is calculated incorrectly when the camera is attached to a node in the scene that has its own transformation (e.g. the camera on the roller coaster).

Further, I update the inverse projection matrix whenever the projection matrix itself is updated, and compute the `fov` property of the perspective camera. There are still other fields of the user camera that are not computed, but I'm not sure how far we want to go with extracting the parameters back out of the projection matrix.

rhuitl committed:
WebXRManager: Correctly update the user camera when it has a parent with a non-identity transform.
Further, calculate the inverse projection matrix and the field of view.

Mugen87 commented:
I'm not sure the issues you are trying to fix with the PR are clear. Do you mind modifying one of the official examples and share a link here so it's possible to see what's going wrong?

rhuitl commented:
I should say that I'm not 100% sure what WebXRManager's idea of the interaction of the user camera and the VR camera is. I assume that:
* the user camera pose should be ignored completely for rendering
* and that the pose of the user camera is set by WebXRManager's when a frame is rendered, so that it matches the VR camera's pose (so that applications can use the user camera pose for their own needs)
* and that the only information that is taken from the user camera is the parent node, in order to make the VR camera move along with the parent node.

I modified the roller coaster example to illustrate the problem: https://github.com/rhuitl/three.js/commit/147b73413eb1724798b6cd61f6be2116febdd8ef

Basically I'm setting the user camera position to something large (100) and expect it to be ignored. Before my patch I get this output:

```
Old:
train.pos    (mworld) 27.93, 13.00, 4.14
train.pos    (matrix) 27.93, 13.00, 4.14
train.pos    (vector) 27.93, 13.00, 4.14
vrcamera.pos (mworld) 27.95, 12.99, 4.14
vrcamera.pos (matrix) -0.04, -0.01, -0.05
vrcamera.pos (vector) 27.95, 12.99, 4.14
camera.pos   (mworld) 12.45, 110.97, 104.95  <-- wrong
camera.pos   (matrix) -0.04, -0.01, -0.05
camera.pos   (vector) -0.04, -0.01, -0.05
```

And afterwards:
```
train.pos    (mworld) 30.13, 12.49, 5.02
train.pos    (matrix) 30.13, 12.49, 5.02
train.pos    (vector) 30.13, 12.49, 5.02
vrcamera.pos (mworld) 30.17, 12.48, 5.06
vrcamera.pos (matrix) -0.01, -0.00, -0.09
vrcamera.pos (vector) 30.17, 12.48, 5.06
camera.pos   (mworld) 30.17, 12.48, 5.06  <-- correct
camera.pos   (matrix) 0.02, -0.00, -0.06
camera.pos   (vector) 0.02, -0.00, -0.06
```

This is the main problem this PR addresses. There are further, smaller improvements:
* The projection matrix was copied from the XR API into the individual array cameras. As `Camera` also has the inverse projection matrix, I added code to compute it - for good measure only, it didn't fix anything for me.
* The same was the case for the VR camera - projection matrix is set, but the inverse was not updated. Now it is.
* Note that `setProjectionFromUnion()` moves the VR camera a bit to the right, but the user camera's position was set before that method call. I rearranged the order so that it is now copying the position AFTER the VR camera has been shifted. That means the user camera was placed at the left eye before, and now it is placed in the center where the VR camera is.
* The user camera's projection matrix was not updated at all, so any code that would load data based on the user camera's frustum was not using the XR system's frustum. Now it is set to the VR camera's projection matrix. Because I had to deal with other library code that relies on the `fov` field I also add code to compute that.

rhuitl commented:
@zalo in https://github.com/mrdoob/three.js/pull/21964#issuecomment-901442113 you mentioned you have an application that the previous pull request broke. Could you test the changes in this PR or tell me how to verify it's not breaking your use case?

zalo commented:
Yep; this is the application: https://github.com/leapmotion/LeapShape

Hopefully I’ll be able to get it tested in the next few days.

rhuitl commented:
I just tried your application with the patch and it seems to work fine, i.e. I can pan the scene by holding the button down and moving the controller around.

rhuitl commented:
@elalish would you be able to check if the changes from this PR allow you to drop the code in your application that you mentioned in https://github.com/mrdoob/three.js/pull/21964#issuecomment-902074459?

elalish commented:
Sorry for the delay! Yes, I just pulled your branch and tested it; everything seems to be working just fine. @mrdoob, this looks safe to merge to me.
